tests/http/sensors/test_http.py

```diff
@@ -54,29 +53,25 @@ def test_http_run_now_sensor_async(self, mock_poke):
 
         assert isinstance(exc.value.trigger, HttpTrigger), "Trigger is not a HttpTrigger"
 
-    def test_http_response_check_does_not_run_async(self):
-        """Asserts that a task is not deferred when response_check arg is passed to HttpSensorAsync."""
-
+    @mock.patch("astronomer.providers.http.sensors.http.HttpSensorAsync.defer")
+    @mock.patch("airflow.sensors.base.BaseSensorOperator.execute")
+    def test_sensor_not_defer(self, mock_execute, mock_defer):
         task = HttpSensorAsync(
             task_id="run_now",
             endpoint="test-endpoint",
             response_check=lambda response: "httpbin" in response.text,
         )
-
-        with mock.patch(
-            "astronomer.providers.http.sensors.http.HttpSensorAsync.defer"
-        ) as mock_defer, mock.patch("airflow.sensors.base.BaseSensorOperator.execute"):
-            task.execute({})
-
```

Comment:
Can we not use mocking it in the same fashion?

Reply:
I prefer to have have small test and one test testing one path here